### PR TITLE
strict tsconfig checks in place for binding-file

### PR DESCRIPTION
**Pull request by danielpeintner**
relevant for to https://github.com/eclipse-thingweb/node-wot/issues/758

### CODE REVIEW

File: packages/binding-file/src/file-client.ts [hunk 1]
```diff
@@ -86,8 +86,9 @@ export default class FileClient implements ProtocolClient {
         error?: (error: Error) => void,
         complete?: () => void
     ): Promise<Subscription> {
-        error(new Error(`FileClient does not implement subscribe`));
-        return null;
+        return new Promise<Subscription>((resolve, reject) => {
```
Comment (relu91) [inline packages/binding-file/src/file-client.ts:89]: I prefer throwing directly. 🤔 is it possible?

File: packages/binding-file/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
     "extends": "../../tsconfig.json",
     "compilerOptions": {
+        "strict": true,
+        "strictFunctionTypes": true,
```
Comment (relu91) [inline packages/binding-file/tsconfig.json:5]: I am doing some changes for other packages, is `strictFunctionTypes` required here? or is it sufficient to set `strict` to true?

**Review by relu91 — COMMENTED** [review]
Great thank you, I have two comments before approving. One side note, I'm working on  `binding-http` from about 80 errors I'm down to six. Let's see if I can provide a PR for this afternoon.

File: packages/binding-file/src/file-client.ts (hunk 1 above)
Comment (danielpeintner) [reply packages/binding-file/src/file-client.ts:89]: Not sure what you mean by "throwing directly" ?

I just aligned this method with what is done in all the other methods in this class
see https://github.com/eclipse-thingweb/node-wot/blob/master/packages/binding-file/src/file-client.ts

File: packages/binding-file/tsconfig.json (hunk 2 above)
Comment (danielpeintner) [reply packages/binding-file/tsconfig.json:5]: I think what I would like to see in the end is a _global_ tsconfig.json file used for all packages. 
Hence I took the same settings that are currently used in `core` to not getting any surprise later (and yes, `strictFunctionTypes `might not be required here )

https://github.com/eclipse-thingweb/node-wot/blob/b25d863ab85774570bf203e126670d2975a0aa77/packages/core/tsconfig.json#L4-L5

File: packages/binding-file/src/file-client.ts (hunk 1 above)
Comment (relu91) [reply packages/binding-file/src/file-client.ts:89]: I mean in theory you don't need to create the promise object and reject it. You can `throw new Error("...")` directly. But I see know that you used the same pattern that was in the file. So for consistency, I'm ok with this, we can fix it later.

File: packages/binding-file/tsconfig.json (hunk 2 above)
Comment (relu91) [reply packages/binding-file/tsconfig.json:5]: Checked the [doc](https://www.typescriptlang.org/tsconfig/#strict):

> Turning this on is equivalent to **enabling all** of the strict mode family options, which are outlined below.

So yeah in a followup PR we should remove this one here and the one in core (or everywhere else).

**Pull request merged**